docs/spec/ibc/queues.md

```diff
@@ -0,0 +1,160 @@
+## 3 Messaging Queue
+
+([Back to table of contents](specification.md#contents))
+
+Messaging in distributed systems is a deeply researched field and a primitive upon which many other systems are built. We can model asynchronous message passing, and make no timing assumptions on the communication channels. By doing this, we allow each zone to move at its own speed, unblocked by any other zone, but able to communicate as fast as the network allows at that moment.
+
+Another benefit of using message passing as our primitive, is that the receiver decides how to act upon the incoming message. Just because one zone sends a message and we have an IBC connection with this zone, doesn't mean we have to execute the requested action. Each zone can add its own business logic upon receiving the message to decide whether to accept or reject the message. To maintain consistency, both sides must only agree on the proper state transitions associated with accepting or rejecting.
```

Comment:
"Just because one zone sends a message and we have an IBC connection with this zone, doesn't mean we have to execute the requested action. Each zone can add its own business logic upon receiving the message to decide whether to accept or reject the message." I am not sure about the semantic that is being implied by this sentence as it consider the content of IBC message to be an action. Is IBC payload agnostic or not?

Reply:
Clarified in several places that this spec is payload-agnostic.